core/src/com/unciv/ui/multiplayer/AddFriendScreen.kt

```diff
@@ -0,0 +1,60 @@
+package com.unciv.ui
+
+import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.scenes.scene2d.ui.Table
+import com.badlogic.gdx.scenes.scene2d.ui.TextField
+import com.unciv.models.translations.tr
+import com.unciv.ui.multiplayer.FriendList
+import com.unciv.ui.pickerscreens.PickerScreen
+import com.unciv.ui.popup.ToastPopup
+import com.unciv.ui.utils.*
+
+class AddFriendScreen(backScreen: ViewFriendsListScreen) : PickerScreen(){
+    init {
+        val friendNameTextField = TextField("", skin)
+        val pasteGameIDButton = "Paste playerID from clipboard".toTextButton()
+        val playerIDTextField = TextField("", skin)
+        val friendlist = FriendList()
+
+
+        topTable.add("Friend name".toLabel()).row()
+        topTable.add(friendNameTextField).pad(10f).padBottom(30f).width(stage.width/2).row()
+
+        pasteGameIDButton.onClick {
+            playerIDTextField.text = Gdx.app.clipboard.contents
+        }
+
+        topTable.add("PlayerID".toLabel()).row()
+        val gameIDTable = Table()
+        gameIDTable.add(playerIDTextField).pad(10f).width(2*stage.width/3 - pasteGameIDButton.width)
+        gameIDTable.add(pasteGameIDButton)
+        topTable.add(gameIDTable).padBottom(30f).row()
+
+        //CloseButton Setup
+        closeButton.setText("Back".tr())
+        closeButton.onClick {
+            backScreen.game.setScreen(backScreen)
+        }
+
+        //RightSideButton Setup
+        rightSideButton.setText("Add friend".tr())
+        rightSideButton.enable()
+        rightSideButton.onClick {
+            friendlist.addNewFriend(friendNameTextField.text, playerIDTextField.text)
+            if (friendlist.addFriendErrorType == "name") {
+                ToastPopup("Friend name is already in your friends list!", this)
+            } else if (friendlist.addFriendErrorType == "id") {
+                ToastPopup("Player ID is already in your friends list!", this)
+            } else if (friendlist.addFriendErrorType == "noName") {
+                ToastPopup("You have to write a name for your friend!", this)
+            } else if (friendlist.addFriendErrorType == "noID") {
+                ToastPopup("You have to write an ID for your friend!", this)
+            } else {
+                backScreen.game.setScreen(backScreen)
+                backScreen.refreshFriendsList()
+            }
```

Comment:
Should I replace with a when statement ?

Reply:
Studio should automatically tell you to, and even do it for you with those little lightbulbs... You could probably even have only one ToastPopup call with something like `val errorMessave = when (friendlist.addFriendErrorType) { Name -> "Hey that dude's already in here!"... else -> { do and exit } }; Toast(errorMessave...` by no means a request, only food for thought. And that sting code collection reeks like a candidate for a local enum...